solana-programs/anchor/audius-data/programs/audius-data/src/lib.rs

```diff
@@ -35,17 +37,23 @@ pub mod audius_data {
         ctx: Context<InitializeUser>,
         base: Pubkey,
         eth_address: [u8; 20],
-        _handle_seed: [u8; 16],
+        handle_seed: [u8; 16],
         _user_bump: u8,
         metadata: String,
     ) -> ProgramResult {
         msg!("Audius::InitUser");
         // Confirm that the base used for user account seed is derived from this Audius admin storage account
         let (derived_base, _) = find_program_address_pubkey(ctx.accounts.admin.key(), ctx.program_id);
-
         if derived_base != base {
             return Err(ErrorCode::Unauthorized.into());
         }
+
+        // Confirm that the derived pda from base is the same as the user storage account
+        let (derived_pda, _) = Pubkey::find_program_address(&[&derived_base.to_bytes()[..32], &handle_seed], ctx.program_id);
```

Comment:
let's make the derived target variable name a bit more explicit here like derived_user_acc or something